test/unit/asset_test.js

```diff
@@ -279,104 +277,121 @@ describe('Asset', function () {
 
-      expect(StellarBase.Asset.compare(assetARST, assetARST)).to.eq(0);
-      expect(StellarBase.Asset.compare(assetARST, assetUSDX)).to.eq(-1);
+      expect(Asset.compare(assetARST, assetARST)).to.eq(0);
+      expect(Asset.compare(assetARST, assetUSDX)).to.eq(-1);
 
-      expect(StellarBase.Asset.compare(assetUSDX, assetARST)).to.eq(1);
-      expect(StellarBase.Asset.compare(assetUSDX, assetUSDX)).to.eq(0);
+      expect(Asset.compare(assetUSDX, assetARST)).to.eq(1);
+      expect(Asset.compare(assetUSDX, assetUSDX)).to.eq(0);
 
       // uppercase should be smaller
-      const assetLower = new StellarBase.Asset('aRST', assetARST.getIssuer());
-      expect(StellarBase.Asset.compare(assetARST, assetLower)).to.eq(-1);
-      expect(StellarBase.Asset.compare(assetLower, assetA)).to.eq(1);
+      const assetLower = new Asset('aRST', assetARST.getIssuer());
+      expect(Asset.compare(assetARST, assetLower)).to.eq(-1);
+      expect(Asset.compare(assetLower, assetA)).to.eq(1);
     });
 
     it('test if asset issuers are being validated as assetIssuerA < assetIssuerB', function () {
-      const assetIssuerA = new StellarBase.Asset(
+      const assetIssuerA = new Asset(
         'ARST',
         'GB7TAYRUZGE6TVT7NHP5SMIZRNQA6PLM423EYISAOAP3MKYIQMVYP2JO'
       );
-      const assetIssuerB = new StellarBase.Asset(
+      const assetIssuerB = new Asset(
         'ARST',
         'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
       );
 
-      expect(StellarBase.Asset.compare(assetIssuerA, assetIssuerB)).to.eq(-1);
-      expect(StellarBase.Asset.compare(assetIssuerA, assetIssuerA)).to.eq(0);
+      expect(Asset.compare(assetIssuerA, assetIssuerB)).to.eq(-1);
+      expect(Asset.compare(assetIssuerA, assetIssuerA)).to.eq(0);
 
-      expect(StellarBase.Asset.compare(assetIssuerB, assetIssuerA)).to.eq(1);
-      expect(StellarBase.Asset.compare(assetIssuerB, assetIssuerB)).to.eq(0);
+      expect(Asset.compare(assetIssuerB, assetIssuerA)).to.eq(1);
+      expect(Asset.compare(assetIssuerB, assetIssuerB)).to.eq(0);
     });
 
     it('test if codes with upper-case letters are sorted before lower-case letters', function () {
       // All upper-case letters should come before any lower-case ones
-      const assetA = new StellarBase.Asset(
+      const assetA = new Asset(
         'B',
         'GA7NLOF4EHWMJF6DBXXV2H6AYI7IHYWNFZR6R52BYBLY7TE5Q74AIDRA'
       );
-      const assetB = new StellarBase.Asset(
+      const assetB = new Asset(
         'a',
         'GA7NLOF4EHWMJF6DBXXV2H6AYI7IHYWNFZR6R52BYBLY7TE5Q74AIDRA'
       );
 
-      expect(StellarBase.Asset.compare(assetA, assetB)).to.eq(-1);
+      expect(Asset.compare(assetA, assetB)).to.eq(-1);
+    });
+  });
+
+  describe('contractId()', function () {
+    it('creates the correct contract IDs', function () {
+      [
+        [
+          Asset.native(),
+          'CB64D3G7SM2RTH6JSGG34DDTFTQ5CFDKVDZJZSODMCX4NJ2HV2KN7OHT'
+        ]
+        // TODO: Add more
```

Comment:
Test-case for alpha-num4 asset:

```suggestion
        [
          Asset.native(),
          'CB64D3G7SM2RTH6JSGG34DDTFTQ5CFDKVDZJZSODMCX4NJ2HV2KN7OHT'
        ],
        [
          new Asset('USD', 'GCP2QKBFLLEEWYVKAIXIJIJNCZ6XEBIE4PCDB6BF3GUB6FGE2RQ3HDVP'),
          'CCWNZPARJG7KQ6N4BGZ5OBWKSSK4AVQ5URLDRXB4ZJXKGEJQTIIRPAHN'
        ]
```

Reply:
perfect, thanks!